alacritty_terminal/src/term/mod.rs

```diff
@@ -459,8 +463,10 @@ impl<'a> Iterator for RenderableCellsIter<'a> {
                     if self.cursor_style == CursorStyle::Block {
                         std::mem::swap(&mut cell.bg, &mut cell.fg);
 
-                        if let Some(color) = self.config.colors.cursor.text {
-                            cell.fg = color;
+                        if self.config.colors.cursor.text.is_some() {
+                            // `ansi::NamedColor::CursorText` is in sync with `cursor.text` color
+                            // from a config.
+                            cell.fg = *self.colors.index(ansi::NamedColor::CursorText);
                         }
```

Comment:
Is there any way to change this? Or are we just storing it in our colors without any real benefit?

Reply:
I didn't find anyway to change it, and the `alacritty_terminal/src/ansi.rs` doesn't have any handler for it atm. Just searching the web result in nothing and `nvim` can change only background color, and not foreground one. So, It seems to me like there is no way to change it. However we can introduce some `proprietary` sequence to change it.

Reply:
I'd be surprised if there was a way to change it and I'm not familiar with it. I'm strongly against introducing a proprietary escape, but we might be able to remove it from colors if we don't need it there.

It's a bit unergonomic to have to load things from two different places. We should just be able to `let Some` it and be good with it.

Reply:
> It's a bit unergonomic to have to load things from two different places. We should just be able to let Some it and be good with it.

You mean returning old cursor colors supporting methods ( You've removed them during rework)?

Reply:
I mean it's unergonomic to check the config colors and then load it from a completely different place, even though they're always the exact same. For someone just looking at that part of the code that makes absolutely no sense at all.

I think I was actually able to change it from the old way to simplify things a bit more. Generally it seems like the `NamedColor::CursorText` just isn't necessary anymore. If this is the only place where we're using it, I think it might be cleaner just to remove it.

Reply:
> I mean it's unergonomic to check the config colors and then load it from a completely different place, even though they're always the exact same.

I've understood this in the first place, I mean, you've removed the following method on purpose.
```rust
impl Config {
...
/// Cursor background color                                             
#[inline]                                                              
pub fn cursor_cursor_color(&self) -> Option<Color> {                   
    self.colors.cursor.cursor.map(|_| Color::Named(NamedColor::Cursor))
}
...                                                                      
}
```
So, I want to bring it back. So cursor color setting will change to:
```rust
if let Some(cursor) = self.config.cursor_cursor_color() {
    renderable_cell.fg = RenderableCell::compute_bg_rgb(self.colors, cursor);
}
```

I think removing `NamedColor::CursorText` is a good idea. So, should I add the following method to the `struct Config` to mimic method suggested above, or access it directly, like it's now?

```rust
#[inline]
pub fn cursor_text_color(&self) -> Option<Rgb> {                   
    self.colors.cursor.text
}
```

Reply:
Sure, looks good to me. One concern might be that one of these is returning a Color and the other Rgb, but I personally wouldn't worry about that too much.